client/src/mvc/library/library-dataset-view.js

```diff
@@ -478,7 +478,7 @@ var LibraryDatasetView = Backbone.View.extend({
                         </button>
                     <% } %>
                     <% if (item.get("can_user_manage")) { %>
-                        <a href="<% rootPath %>/library/folders/permissions/<%- item.get("folder_id") %>/dataset/<%- item.id %>">
+                        <a href="<% rootPath %>/libraries#/folders/permissions/<%- item.get("folder_id") %>/dataset/<%- item.id %>">
```

Comment:
I would much rather use history mode in the routers and avoid breaking URLs into hashed fragments

Reply:
well, then we would have to define each individual route in 3 different places: vue-router itself, `AnalysisRouter` and `buildapp.py`. The PR removes it [here](https://github.com/galaxyproject/galaxy/pull/11614/files#diff-0d8ac94241a4666f6f926894cd7a018d670cdce28b5b50e67a0328c8f6e7e823) and  [here](https://github.com/galaxyproject/galaxy/pull/11614/files#diff-edaf3dcda3c30261c3bbe2fa65e7c5fd0e800f0ddb1e5046d9d1a46e6dab1eea). 
IMHO it looks more consistent, besides it the approach was also used in the previous implementation.

Once the global router is in place, we can add merge library routes with great application-wide Vue router in history mode.

Reply:
I know it removes individual endpoints -- I'm saying I think you can go ahead and do that here and avoid `#` in URLs, which we have tried hard to do for years.

Reply:
> I know it removes having to individually declare all the endpoints -- I'm saying I think you can go ahead and do that here _and_ avoid `#` in URLs, which we have tried hard to do for years.

why are we so against `#`? what is the motivation behind it? 
If it's just aesthetics, I don't really see a problem it being at least a temporary solution, until we have Vue router defined globally. Defining same things 5 times in 3 different places, doesn't look right.

Reply:
I am clearly not arguing for defining the same things in multiple places.  As I have said, I think we can just do this correctly, now.  I'll make a pass over this later.